### PR TITLE
[PPP-3455] - Dynamic code injection vulnerabilities found

### PR DESCRIPTION
**Pull request by akhayrutdinov**
- replace `eval()` with `JSON.parse()` in models-svc.js, ref.js

@rfellows, @krivera-pentaho, @diogofscmariano, review it please

### CODE REVIEW

File: package-res/resources/web/dataapi/models-svc.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ pentaho.pda.SvcHandler.prototype.getSources = function(callback, options) {
             var contextName = options['context'];
             var query = "domainName=&"+((contextName)? 'context='+contextName : "");
             var jsonStr = pentahoPost( url, query, null, 'text/text' );
```
Comment (rfellows) [inline package-res/resources/web/dataapi/models-svc.js]: I would prefer that we move away from pentahoPost and to the dojo request method. you can specify the return type and not have to parse it after the fact.

Comment (akhayrutdinov) [inline package-res/resources/web/dataapi/models-svc.js]: @rfellows, I'm afraid it is a bit wider, than expected from this PR. That is not a problem for me to introduce such changes, but it misleads from the focus of the ticket.

@pamval, @mbatchelor, do you agree with Rob (replacement of `pentahoPost`)

Comment (pamval) [inline package-res/resources/web/dataapi/models-svc.js]: I'd go ahead with that change here, in this call in this file. 
Not a global replacement of pentahoPost everywhere.

Comment (krivera-pentaho) [inline package-res/resources/web/dataapi/models-svc.js]: Would we want to consider replacing the implementation of pentahoPost to use the dojo request method?

Comment (rfellows) [inline package-res/resources/web/dataapi/models-svc.js]: no, there are probably tons of uses of it out there still and we'd have to make it return exactly what it returns now, which is potentially still using eval, so it wouldn't really provide the whole solution without refactoring all of the places that use it as well.

**Comment by akhayrutdinov** [comment]
@rfellows, I've updated the PR with using `dojo/request`. Can you please review it? I haven't found any examples of using `request.post()` in the project, thus I'm not sure I migrated from the old API correctly

File: package-res/resources/web/dataapi/models-svc.js [hunk 2]
```diff
@@ -54,8 +54,20 @@ pentaho.pda.SvcHandler.prototype.getSources = function(callback, options) {
             var url = this.SERVICE_URL+'?action=listmodels';
             var contextName = options['context'];
             var query = "domainName=&"+((contextName)? 'context='+contextName : "");
-            var jsonStr = pentahoPost( url, query, null, 'text/text' );
-            var rawModels = eval( '('+jsonStr+')' );
+
+            var rawModels;
+            request.post(url, {
```
Comment (rfellows) [inline package-res/resources/web/dataapi/models-svc.js]: If we aren't sending anything in the data portion of the request, why bother with post? wouldn't a get suffice?

Comment (akhayrutdinov) [inline package-res/resources/web/dataapi/models-svc.js]: @rfellows, I wish I knew the answer :smile: 
That is how is was implemented, I'm surprised with POST too

Comment (rfellows) [inline package-res/resources/web/dataapi/models-svc.js]: ok, i'll merge it then

**Pull request merged**